php/tests/generated_class_test.php

```diff
@@ -679,6 +680,40 @@ public function testNamespaceMessage()
     }
 
     #########################################################
+    # Test across different packages and php namespaces
+    #########################################################
+
+    public function testNamespaces()
```

Comment:
How should I test for this? If I do `class_exists('\TestNamespaces');`, this will return false. If I define `\TestNamespaces`, I will get an import error because I am importing that classname at the top of the test.

I could define a class `\TestNamespaces` in this file, and import `\Foo\TestNamespaces as FooTestNamespaces`. If this is what you're testing for, that's implemented on a language level, and PHP will not accept `\[classname]` for `[classname]` in a file with a namespace.